### PR TITLE
feature: add deeplink capability to bob

### PR DESCRIPTION
**Pull request by chikeichan**
This PR allow others to deeplink to bob using `bob://` protocol name. 

The two deeplink URLs that are implemented now are: 

- `bob://openname?name=test2` -- open auction page
- `bob://openmanager?name=test2` -- open domain manager page
- `bob://updaterecord?name=test2&txt=hello` -- 1-click update record

**Expected Deeplink Behavior**
- when opening a deeplink on a *locked* wallet, bob will allow user to choose a wallet, login, then redirect to the deeplink
- when opening a deeplink on an *unlocked* wallet, bob will redirect to the deeplink

Also as part of the PR, Domain Manager view is updated to disable all inputs/action when user is not the owner.

deeplink

deeplink2

### CODE REVIEW

**Comment by pinheadmz** [comment]
Interesting. As long as you're playing around with this, maybe Bob could support [BIP21](https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki)-style URIs for payments as well. That will also be more generic and can be standardized across the ecosystem with our own ["HIP"](https://github.com/handshake-org/HIPs)

example:

```
handshake://hs1qde7jaw6qgzzfu83upn3twvsyhh0zrshg76qe0x?amount=1.123456
```

**Comment by chikeichan** [comment]
> Interesting. As long as you're playing around with this, maybe Bob could support [BIP21](https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki)-style URIs for payments as well. That will also be more generic and can be standardized across the ecosystem with our own ["HIP"](https://github.com/handshake-org/HIPs)
> 
> example:
> 
> ```
> handshake://hs1qde7jaw6qgzzfu83upn3twvsyhh0zrshg76qe0x?amount=1.123456
> ```

I did not know that there is a BIP for URI schemas -- thanks!!

For this particular use case I am just looking to unblock a way to deeplink Bob to the Domain Info and DNS Record screen. Since the use case is fairly limited right now, i will keep this Bob-specific for now, and will keep an eye out for timing to submit/integrate to Handshake's own version of BIP21.

File: app/index.js [hunk 1]
```diff
@@ -19,6 +23,69 @@ ipcRenderer.on('ipcToRedux', (_, message) => {
...
+        store.dispatch(setDeeplinkParams({ txt }));
+      }
+
+      if (isLocked) {
+        store.dispatch(setDeeplink(message));
+      } else {
+        history.push(`/domain_manager/${name}`);
+      }
+      return;
+    case "//openauction":
```
Comment (sdtsui) [inline app/index.js:56]: Would it make more sense to have `openauction` be `openname`, to match more closely with the ecosystem's block explorers?   i.e. The on-chain auction/ownership page is the high level view of the name, so you're opening 
 (information for) the name.

Same line of thinking: `opendomain` could be `openmanager` or `namemanager` as name itself would be implied by a param.  I'm not sure if you'd prefer verb-prefixes, so maybe `openmanager` is best.

File: app/main.js [hunk 2]
```diff
@@ -24,6 +24,14 @@ if (
   require('electron-debug')();
 }
 
+app.setAsDefaultProtocolClient('bob');
+
+// Protocol handler for osx
```
Comment (sdtsui) [inline app/main.js:29]: Is this osx only for now?  Ping me on Signal and I can test Windows whenever it's ready.

**Review by sdtsui — COMMENTED** [review]
Main flows QA'd with no issues.  Left two questions @chikeichan .

File: app/index.js (hunk 1 above)
Comment (chikeichan) [reply app/index.js]: updated

File: app/main.js (hunk 2 above)
Comment (chikeichan) [reply app/main.js]: just added support for window as well

**Pull request merged**